code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -23,6 +23,15 @@ datum
 				for(var/path in paths)
 					var/datum/reagent/D = new path()
 					chemical_reagents_list[D.id] = D
+			if(!safe_reagents_list)
+				//Filtered list to prevent shit like xenobotany spitting out 200 bar drinks
+				var/paths = typesof(/datum/reagent) - /datum/reagent
+				safe_reagents_list = list()
+				for(var/path in paths)
+					var/datum/reagent/D = new path()

```

Comment:
Actually, if you're going to make a list of all reagent datums, do it ONCE in global_lists.dm and then call that.